lua/smart-splits/mux/kitty.lua

```diff
@@ -97,6 +97,22 @@ function M.resize_pane(direction, amount)
   return ok
 end
 
+function M.on_init()
+  vim.api.nvim_create_autocmd('VimEnter', {
+    group = vim.api.nvim_create_augroup('KittySetVarVimEnter', { clear = true }),
+    callback = function()
+      io.stdout:write('\x1b]1337;SetUserVar=IS_NVIM=MQo\007')
+    end,
+  })
+
+  vim.api.nvim_create_autocmd('VimLeave', {
+    group = vim.api.nvim_create_augroup('KittyUnsetVarVimLeave', { clear = true }),
+    callback = function()
+      io.stdout:write('\x1b]1337;SetUserVar=IS_NVIM\007')
+    end,
+  })
```

Comment:
One other thing I forgot to mention; you shouldn't need to manually create the autocmds here. The way the API works is, the mux backend should provide an `on_init` and an `on_exit`, and the shared startup procedure handles creating the autocmds. This way `VimSuspend` and `VimResume` are handled automatically as well.

https://github.com/mrjones2014/smart-splits.nvim/blob/95d11bfc83263236431ead7dc017f5ab0ff992ca/lua/smart-splits/mux/utils.lua#L46-L78

So basically, you should just have:

```lua
function M.on_init()
  io.stdout:write('\x1b]1337;SetUserVar=IS_NVIM=MQo\007')
end

function M.on_exit()
  io.stdout:write('\x1b]1337;SetUserVar=IS_NVIM\007')
end
```

Reply:
Ah! I sort of hastily added them without really exploring the rest of the code. Fixed now!